### PR TITLE
Upgrade to Golang 1.20

### LINKED ISSUE #296: Upgrade to latest Go 1.20

### PR DESCRIPTION
**Pull request by ekampf**
Resolves #296

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/Twingate/terraform-provider-twingate
 
-go 1.19
+go 1.20
```
Comment (vmanilo) [inline go.mod:3]: @ekampf I guess in github workflows we also would need to updated go version

Comment (ekampf) [reply go.mod:3]: @vmanilo setup-go reads it from go.mod

Comment (ekampf) [reply go.mod:3]:
```
    - name: Set up Go
      uses: actions/setup-go@v3
      with:
        go-version-file: 'go.mod'
        cache: true
      id: go
```

**Review by liorr — APPROVED** [review]
LGTM

File: go.mod (hunk 1 above)
Comment (vmanilo) [reply go.mod:3]: yeah right, I mean this line

https://github.com/Twingate/terraform-provider-twingate/blob/main/.github/workflows/release.yml#L31

Comment (vmanilo) [reply go.mod:3]: and readme https://github.com/Twingate/terraform-provider-twingate/blob/main/README.md?plain=1#L11

**Review by vmanilo — APPROVED** [review]
looks good 👍

**Pull request merged**